src/chains/ethereum/ethereum/tests/provider.test.ts

```diff
@@ -244,4 +245,40 @@ describe("provider", () => {
       );
     });
   });
+
+  describe("web3 compatibility", () => {
+    const networkId = 1234;
+    let provider: EthereumProvider;
+    let web3: Web3;
+    let accounts: string[];
+
+    beforeEach(async () => {
+      provider = await getProvider({
+        chain: { networkId },
+        wallet: { deterministic: true }
+      });
+      web3 = new Web3();
+      // TODO: remove "as any" once we've fixed our typing issues with web3
+      web3.setProvider(provider as any);
+      accounts = await web3.eth.getAccounts();
+    });
+
+    it("returns things via legacy", async () => {
+      const subscription = web3.eth.subscribe("newBlockHeaders");
+      assert(subscription != null);
+      assert.notStrictEqual(subscription.id, false);
+
+      // if the data isn't properly serialized before emitting, web3 won't
+      // ever catch this
```

Comment:
What do you mean by "catch" here? Would "handle" be a better word?

Reply:
Yes, I definitely wasn't thinking of "catch" as an actual technical term here. Handle is better.
```suggestion
      // ever handle this
```